### PR TITLE
Move to Guzzle 6.0.*

### PR DESCRIPTION
**Pull request by thomasez**
This is what I had to do to get this working with Guzzle 6.

It is not heavily tested! Basically only tested login and GET since I haven't had the opportunity to test more than that yet.

It's more than just guessing on the other functions, but postFile may be BC-broken. The others should work without changes to the code using this wrapper.

### CODE REVIEW

File: src/Spinegar/Sugar7Wrapper/Clients/Guzzle.php [hunk 1]
```diff
@@ -275,33 +318,39 @@ public function get($endpoint, $parameters = array())
   * Function: get()
   * Parameters: 
   *   $endpoint = endpoint per API specs
+<<<<<<< HEAD
```
Comment (mmarum-sugarcrm) [inline src/Spinegar/Sugar7Wrapper/Clients/Guzzle.php]: Looks like some diff tags left over from merge.

**Comment by spinegar** [comment]
Hi @thomasez,

Firstly, great work! This is a great contribution to the project!

That said, I'm all for a move to a newer version of guzzle but there a lot of projects using this lib that we want to make sure dont see a negative impact.  For that reason it's about time that unit tests are added to this lib. Ideally, before we merge in these guzzle 6.x updates.

I'll try and drop in a unit test framework in the next week or so. Feel free to contribute in this area if you are interested.

**Comment by thomasez** [comment]
I can wait for the test framework is in place. I've been using my version of the wrapper for some time now and it works for me :=) (But i am not posting any data to it, so that should be tested..)

**Comment by spinegar** [comment]
@thomasez can you move the 6.x implementation in to its own client in `src/Spinegar/Sugar7Wrapper/Clients/` ? I'm adding unit tests and ultimately think it would be safe to keep the guzzle 3.x implementation available in case someone needs it.

**Comment by thomasez** [comment]
Yup, I'll do that.

(In the UK now, have to get home first. A week or so I guess)

**Comment by thomasez** [comment]
Looking at the (failed) tests I can see that I've forgotten some hack to composer.json.

Should I do "suggest" instead of "require" on both guzzle 3 and 6? Then let the user add their choice of guzzle to their projects composer.json?

(But with that change I guess we create a BC break and can set Guzzle6 as default.)

Or we could do the python thingie with try/catch on the construct with eventual fallback to the other.

**Comment by spinegar** [comment]
Hey @thomasez - the failed check is due to how the tests are configured at the moment. I pre-maturely added travis integration :)

The unit tests require a username and password to be entered to run (see https://github.com/spinegar/sugarcrm7-api-wrapper-class/blob/master/tests/Sugar7Wrapper.php). I haven't found a convenient method for mocking the http requests in PHP. I think the next step here is to stub the requests but that kinda sounds like a pain in the ass. 

What are your thoughts?

**Comment by thomasez** [comment]
I have to admit I am very lazy when it comes to tests. So I have no real clue how to do it.

Especially testing both guzzleversions.

**Comment by spinegar** [comment]
@thomasez I invited you to be a contributor to the project feel free to make updates and merge as needed. Appreciate all of your contributions!

**Comment by thomasez** [comment]
Been trying to move it all to my Guzzle.php for Guzzle6, but have to admit the multiple request / send part is bugging me.

@jlyons-bgl Made the PR/Commit, maybe you can tell us if you still ned it or can hack it back in?

I can make a guzzle6 branch you can look at.

**Comment by spinegar** [comment]
@thomasez  IMO, let's move this forward even if we have to leave behind multiple requests support. It's not a normal use case.

**Comment by thomasez** [comment]
Added a branch with the changes needed for me. I've just tested that that worked for now.

Gotta see if I can run the tests, but now we have something at least.

**Comment by spinegar** [comment]
closing PR since we just merged in guzzle6 support

**Pull request closed without merging**